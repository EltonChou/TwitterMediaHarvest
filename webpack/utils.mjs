/**
 * @param {import('webpack').WebpackOptionsNormalized} argv - Webpack CLI arguments
 * @returns {import('webpack').WebpackOptionsNormalized & { mode: 'production' }}
 */
export const isProduction = argv => argv.mode === 'production'

/**
 * @param {Record<string, boolean | string>} env - Environment variables passed to webpack
 * @returns {string}
 * @throws {Error} If target is not defined or not a string
 */
export const getTarget = env => {
  const target = env.target
  if (!target || typeof target !== 'string') {
    throw new Error('Target must be defined and must be a string')
  }
  return target
}
