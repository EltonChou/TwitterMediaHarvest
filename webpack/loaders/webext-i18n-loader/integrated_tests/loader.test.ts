import type { LoaderOptions } from '../src/loader.js'
import { Volume, createFsFromVolume } from 'memfs'
import path from 'path'
import webpack from 'webpack'

const compiler = (fixture: string, options: LoaderOptions) => {
  const webpackCompiler = webpack({
    context: path.dirname(__filename),
    entry: fixture,
    output: {
      path: path.resolve(path.dirname(__filename)),
      filename: 'bundle.js',
    },
    module: {
      rules: [
        {
          test: /\.ts$/,
          use: [
            { loader: 'ts-loader' },
            {
              loader: path.resolve(process.cwd(), 'dist', 'index.js'),
              options,
            },
          ],
        },
      ],
    },
  })

  webpackCompiler.outputFileSystem = createFsFromVolume(
    new Volume()
  ) as webpack.OutputFileSystem
  webpackCompiler.outputFileSystem.join = path.join.bind(path)

  return new Promise<webpack.Stats | undefined>((resolve, reject) => {
    webpackCompiler.run((err, stats) => {
      if (err) reject(err)
      if (stats && stats.hasErrors()) reject(stats.toJson().errors)

      resolve(stats)
    })
  })
}

test('test', async () => {
  const stats = await compiler(
    path.resolve(path.dirname(__filename), 'fixtures', 'i18n.ts'),
    {
      expressions: ['i18n', /getText/],
    }
  )

  expect(stats).toBeDefined()
  if (stats) {
    const output = stats.toJson({ source: true })?.modules?.at(0)?.source
    expect(output).toMatchSnapshot()
  }
}, 10000)
