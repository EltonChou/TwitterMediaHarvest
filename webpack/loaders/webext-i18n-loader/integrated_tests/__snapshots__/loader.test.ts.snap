// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test 1`] = `
""use strict";
Object.defineProperty(exports, "__esModule", { value: true });
function i18n(msg, context, placeholders) {
    return msg + context || '' + JSON.stringify(placeholders);
}
function nope(msg) {
    return msg;
}
function getText(msg) {
    return msg;
}
i18n("6a30f630de");
i18n("6a30f630de", "9f86d08");
i18n("6a30f630de", "9f86d08", { foo: 'bar' });
nope('kappa');
getText("ea38562ecb");
"
`;
